doc/tutorial/argument_parser/index.md

```diff
@@ -0,0 +1,414 @@
+
+Set an option/flag to advanced if you do not want the option to be displayed in the normal help page (`-h/--help`). Instead, the advanced options are only displayed when calling `-hh/--advanced-help`. This can be helpful if you want to avoid to bloat your help page with too much information for unexperienced users of your application but still provide thorough information on demand.
+
+Set an option/flag to hidden, if you want to completely hide it from the user. It will never appear on the help page nor any export format. For example, this can be useful for debugging reasons.
+
+\assignment{Assignment 5}
+
+Extend the solution from assignment 4 by declaring the `-s/--season` option as required.
+
+Check if your options are set correctly by trying the following call:
+
+```console
+./game_of_parsing -H --aggregate-by median data.tsv
+[Winter has come] Option -s/--season is required but not set.
+```
+\endassignment
+\solution
+\include doc/tutorial/argument_parser/solution5.cpp
+\endsolution
+
+# Validation of (positional) option values
+
+Our applications often do not allow just any value to be passed as input arguments and if we do not check for them the program may run into undefined behaviour. The best way to carefully restrict user input is to directly check the input when parsing the command line. The seqan3::argument_parser provides **validators** for a given (positional) option.
+
+A *validator* is a [functor](https://stackoverflow.com/questions/356950/what-are-c-functors-and-their-uses) that is called within the argument parser after retrieving and converting a command line argument. We provide several validators which we hope cover most of the use cases but you can always create your own validator (see section todo).
+
+\attention You can pass a validator to the *add_option* only after passing the seqan3::option_spec parameter. Pass the seqan3::option_spec::DEFAULT tag if there are no further restrictions on your option.
+
+## SeqAn3 validators
+
+The following validators are provided in the SeqAn3 library and can be included with the following header:
+
+\snippet doc/tutorial/argument_parser/small_snippets.cpp validator_include
+
+\note All the validators below work on single values or a container of values. In case the variable is a container, the validator is called **on each of the values** separately.
+
+### The seqan3::arithmetic_range_validator
+
+On construction, this validator receives a maximum and a minimum number.
+The validator throws a seqan3::parser_invalid_argument exception whenever a given value does not lie inside the given min/max range.
+
+\snippet test/snippet/argument_parser/validators_1.cpp validator_call
+
+Our application has a another flaw that you might have noticed by now: If you supply a season that is not in the data file the program will again misbehave. Instead of fixing the program, let's restrict the user input accordingly.
```

Comment:
```suggestion
Our application has a another flaw that you might have noticed by now: If you supply a season that is not in the data file, the program will again misbehave. Instead of fixing the program, let's restrict the user input accordingly.
```